integration-tests/src/main/java/org/apache/polaris/service/it/test/PolarisRestCatalogIntegrationTest.java

```diff
@@ -1549,4 +1554,412 @@ public void testUpdateTableWithReservedProperty() {
+      v2ETag = v2Response.getHeaders().getFirst(HttpHeaders.ETAG).toString();
+    }
+
+    // Evolve schema to v3 (add age column)
+    restCatalog.loadTable(tableId).updateSchema()
+        .addColumn("age", Types.IntegerType.get())
+        .commit();
+
+    // Load table and get v3 ETag
+    Invocation v3Invocation =
+        catalogApi
+            .request("v1/" + currentCatalogName + "/namespaces/ns1/tables/test_schema_evolution_table")
+            .build("GET");
+
+    String v3ETag;
+    try (Response v3Response = v3Invocation.invoke()) {
+      assertThat(v3Response.getStatus()).isEqualTo(Response.Status.OK.getStatusCode());
+      assertThat(v3Response.getHeaders()).containsKey(HttpHeaders.ETAG);
+      v3ETag = v3Response.getHeaders().getFirst(HttpHeaders.ETAG).toString();
+    }
+
+    // Verify all ETags are different
+    assertThat(v1ETag).isNotEqualTo(v2ETag);
+    assertThat(v1ETag).isNotEqualTo(v3ETag);
+    assertThat(v2ETag).isNotEqualTo(v3ETag);
+
+    // Test If-None-Match with v1 ETag against current v3 table
+    Invocation v1EtagTestInvocation =
+        catalogApi
+            .request("v1/" + currentCatalogName + "/namespaces/ns1/tables/test_schema_evolution_table")
+            .header(HttpHeaders.IF_NONE_MATCH, v1ETag)
+            .build("GET");
+
+    try (Response v1EtagTestResponse = v1EtagTestInvocation.invoke()) {
+      // Should return 200 OK because table has evolved since v1
+      assertThat(v1EtagTestResponse.getStatus()).isEqualTo(Response.Status.OK.getStatusCode());
+      assertThat(v1EtagTestResponse.getHeaders()).containsKey(HttpHeaders.ETAG);
+
+      String currentETag = v1EtagTestResponse.getHeaders().getFirst(HttpHeaders.ETAG).toString();
+      assertThat(currentETag).isEqualTo(v3ETag); // Should match current v3 ETag
+    }
+
+    // Test with multiple ETags including v1 and v2
+    String multipleETags = v1ETag + ", " + v2ETag;
+    Invocation multipleEtagsInvocation =
+        catalogApi
+            .request("v1/" + currentCatalogName + "/namespaces/ns1/tables/test_schema_evolution_table")
+            .header(HttpHeaders.IF_NONE_MATCH, multipleETags)
+            .build("GET");
+
+    try (Response multipleEtagsResponse = multipleEtagsInvocation.invoke()) {
+      // Should return 200 OK because current v3 ETag doesn't match v1 or v2
+      assertThat(multipleEtagsResponse.getStatus()).isEqualTo(Response.Status.OK.getStatusCode());
+    }
+
+    // Test with current v3 ETag
+    Invocation currentEtagInvocation =
+        catalogApi
+            .request("v1/" + currentCatalogName + "/namespaces/ns1/tables/test_schema_evolution_table")
+            .header(HttpHeaders.IF_NONE_MATCH, v3ETag)
+            .build("GET");
+
+    try (Response currentEtagResponse = currentEtagInvocation.invoke()) {
+      // Should return 304 Not Modified because ETag matches current version
+      assertThat(currentEtagResponse.getStatus()).isEqualTo(Response.Status.NOT_MODIFIED.getStatusCode());
```

Comment:
Would it also make sense to test this when `v3ETag` is combined with `v1ETag`?

Reply:
Added another case